File: python/qiniu-lab-python/rs/stat.py
#coding=utf-8
__author__ = 'jemy'

'''
本例子演示如何使用stat接口获取文件的基本信息
文档：http://developer.qiniu.com/docs/v6/api/reference/rs/stat.html
'''

import qiniu
import sys
import time

accessKey = "<Your Access Key>"
secretKey = "<Your Secret Key>"

'''
获取单个文件的基本信息

bucket 空间名
key 文件名
'''


def stat(bucket, key):
    auth = qiniu.Auth(accessKey, secretKey)
    bm = qiniu.BucketManager(auth)
    retData, respInfo = bm.stat(bucket, key)
    if retData != None:
        print("Stat Result:")
        print("  File Size:\t" + str(retData["fsize"]) + " Bytes")
        print("  Hash:\t\t" + retData["hash"])
        print("  MimeType:\t" + retData["mimeType"])
        print("  PutTime:\t" + time.ctime(retData["putTime"] / 10000000))
    else:
        if respInfo.status_code == 401:
            print("Unauthorized error, please set your access key and secret key")
        else:
            print("Error: " + respInfo.error)


def main():
    if len(sys.argv) == 3:
        bucket = sys.argv[1]
        key = sys.argv[2]
        stat(bucket, key)
    else:
        print("Usage: stat.py bucket key")


if __name__ == "__main__":
    main()

File: java/qiniu-lab-java/src/com/qiniulab/rs/Stat.java
package com.qiniulab.rs;

import com.qiniu.common.QiniuException;
import com.qiniu.storage.BucketManager;
import com.qiniu.storage.model.FileInfo;
import com.qiniu.util.Auth;
import com.qiniulab.config.LabConfig;

/**
 * 获取文件的基本信息
 * */
public class Stat {

	public static void stat() {
		String bucket = "if-pbl";
		String key = "qiniu.png";
		Auth auth = Auth.create(LabConfig.ACCESS_KEY,
				LabConfig.SECRET_KEY);
		BucketManager bm = new BucketManager(auth);
		try {
			FileInfo fi = bm.stat(bucket, key);
			System.out.println(fi.hash);
			System.out.println(fi.fsize);
			System.out.println(fi.mimeType);
			System.out.println(fi.putTime);
			System.out.println(fi.endUser);
		} catch (QiniuException ex) {
			// 401 表示 AccessKey, SecretKey 设置不对
			// 612表示文件不存在
			// 631表示空间不存在
			System.err.println(ex.response.statusCode);
			System.err.println(ex.response.error);
		}
	}

	public static void main(String args[]) {
		stat();
	}
}
